src/Platform/PlatformStorage.php

```diff
@@ -0,0 +1,184 @@
+    $this->dispatcher->dispatch(CommonConsoleEvents::GET_PLATFORM_TYPE, $platform_event);
+    return $this->getPlatform($platform_event, $config);
+  }
+
+  /**
+   * Checks to see if a specific platform exists.
+   *
+   * @param string $name
+   *   The platform to check to see if it exists.
+   *
+   * @return bool
+   */
+  public function exists(string $name) : bool {
+    $directory = $this->ensureDirectory();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "$name.yml"]);
+    return $this->filesystem->exists($alias_file);
+  }
+
+  /**
+   * Save a platform object.
+   *
+   * @param \EclipseGc\CommonConsole\PlatformInterface $platform
+   *   The platform to save.
+   *
+   * @return \EclipseGc\CommonConsole\PlatformInterface|null
+   *   Returns a fully loaded platform object on success. Null on failure.
+   */
+  public function save(PlatformInterface $platform) : ?PlatformInterface {
+    $directory = $this->ensureDirectory();
+    $alias = $platform->getAlias();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "{$alias}.yml"]);
+    try {
+      $this->filesystem->dumpFile($alias_file, Yaml::dump($platform->export()));
+      return $this->load($alias);
+    }
+    catch (IOException $e) {}
+  }
+
+  /**
+   * Delete a platform.
+   *
+   * @param \EclipseGc\CommonConsole\PlatformInterface $platform
+   *   The platform to delete.
+   *
+   * @throws \Exception
+   */
+  public function delete(PlatformInterface $platform) : void {
+    $directory = $this->ensureDirectory();
+    $alias_file = implode(DIRECTORY_SEPARATOR, [$directory, "{$platform->getAlias()}.yml"]);
+    if (!$this->filesystem->exists($alias_file)) {
+      // @todo throw custom exception.
+      throw new \Exception(sprintf("The expected alias file was missing from: %s.", $alias_file));
+    }
+    $this->filesystem->remove($alias_file);
+  }
+
+  /**
+   * Instantiates a platform via its factory or through a naive new statement.
+   *
+   * @param \EclipseGc\CommonConsole\Event\GetPlatformTypeEvent $event
+   *   The GetPlatformTypeEvent object.
+   * @param \Consolidation\Config\ConfigInterface $config
+   *   The configuration values to use for instantiating the platform.
+   *
+   * @return \EclipseGc\CommonConsole\PlatformInterface
+   */
+  protected function getPlatform(GetPlatformTypeEvent $event, ConfigInterface $config) {
+    if ($factory = $event->getFactory()) {
+      if ($this->container->has($factory)) {
+        // @todo check the factory is an instance of PlatformFactoryInterface.
+        return $this->container->get($factory)->create($event, $config, $this->runner);
+      }
+      if (class_exists($factory)) {
+        $factory = new $factory();
+        return $factory->create($event, $config, $this->runner);
+      }
+      throw new LogicException(sprintf("Platform factory service id: %s is missing or undefined.", $factory));
+    }
+    $class = $event->getClass();
+    return new $class($config, $this->runner);
```

Comment:
Is it a crazy idea, if we'd have one factory called PlatformFactory and it would be responsible for instantiating platforms of a given type? This logic here seems to me an extension of a factory pattern wrapping the existing one, which could be moved to PlatformFactory and all that could be encapsulated in one place. Also at the end of the day we wouldn't need to have factories in acquia-contnethub-cli at all, because all of these have been taken care of in the common-console.

Reply:
I agree. It is something I was thinking about.

Reply:
Agreed. Created a PlatformFactory object. This object takes Config and turns it into a Platform one way or another.